raster/src/main/scala/geotrellis/raster/ConstantTile.scala

```diff
@@ -243,6 +249,46 @@ object ConstantTile {
       case ct: FloatCells => FloatConstantTile.fromBytes(bytes, cols, rows, ct)
       case ct: DoubleCells => DoubleConstantTile.fromBytes(bytes, cols, rows, ct)
     }
+
+  /**
+   * Create a [[ConstantTile]] that is set to the nodata value of the given celltype.
+   *
+   * @param   t      The [[CellType]] of the new [[ConstantTile]], it should have a NoData value.
+   * @param   cols   The number of columns that the new [[ConstantTile]] should have
+   * @param   rows   The number of rows that the new [[ConstantTile]] should have
+   * @return         The new [[ConstantTile]]
+   */
+  def emptyTile(t: CellType, cols: Int, rows: Int): Tile = {
+    if(t.isInstanceOf[NoNoData]) {
+      throw new IllegalArgumentException(s"Can not construct an empty constant tile for a celltype without nodata handling: $t")
+    }else{
+      t match {
+        case _: BitCells => BitConstantTile(false, cols, rows)
+        case ct: ByteUserDefinedNoDataCellType => ByteConstantTile(ct.noDataValue ,cols, rows, ct)
+        case ct: ByteCells => ByteConstantTile(t.asInstanceOf[HasNoData[Byte]].noDataValue ,cols, rows, ct)
```

Comment:
I think we can capture all of the cases here with e.g. [these types](https://github.com/locationtech/geotrellis/blob/2cc473aaa58c544651fd0e541ae4af0fe0c1c34a/raster/src/main/scala/geotrellis/raster/CellType.scala#L347-L350) and avoid casting altogether. Should be safe since you explicitly check above to make sure the `CellType` extends `HasNoData`

An example of the change might be something like this:
before:
```
case ct: ByteCells => ByteConstantTile(t.asInstanceOf[HasNoData[Byte]].noDataValue ,cols, rows, ct)
```

After:
```
case ct: ByteConstantNoDataCellType => ByteConstantTile(ct.noDataValue ,cols, rows, ct)
```

Reply:
Ok, makes sense. Committed a first attempt, but had a local compile issue that I don't get yet, so to be seen if CI manages to build it.

Reply:
@moradology  it also failed in CI. I created this code some time ago, and am guessing that I also resorted to the casting because I couldn't find a better solution. Can you perhaps tell if it can be solved more elegantly?

Reply:
Also probably imports are in place, and the syntax is not quite there, these are singletons:

```scala
case ct @ ByteConstantNoDataCellType => ByteConstantTile(ct.noDataValue ,cols, rows, ct)
case ct @ UByteConstantNoDataCellType => UByteConstantTile(ct.noDataValue ,cols, rows, ct)
```

Reply:
Thanks, I'm discovering new Scala things here. What eventually worked for me was using `ByteConstantNoDataCellType.type`. The `@` didn't seem to work, but I probably missed something.